bliss/surveys/sdss.py

```diff
@@ -2,88 +2,97 @@
-        # take a way to specify labels? e.g., from sdss PhotoFullCatalog or decals catalog
     ):
-        super().__init__()
+        super().__init__(predict_device, predict_crop)
 
         self.sdss_path = Path(sdss_dir)
         self.rcfgcs = []
 
-        self.run = run
-        self.camcol = camcol
-        self.fields = fields
-        self.bands = [0, 1, 2, 3, 4]
-
-        self.downloader = SDSSDownloader(run, camcol, fields[0], download_dir=sdss_dir)
+        self.sdss_fields = sdss_fields
+        self.bands = SloanDigitalSkySurvey.BAND_IDXS
 
         self.prepare_data()
-        assert self.items is not None, "No data found even after prepare_data()."
-
-        self.predict_device = predict_device
-        self.predict_crop = predict_crop
 
-    def prepare_data(self):
-        pf_file = f"photoField-{self.run:06d}-{self.camcol:d}.fits"
-        camcol_path = self.sdss_path.joinpath(str(self.run), str(self.camcol))
-        pf_path = camcol_path.joinpath(pf_file)
-        if not Path(pf_path).exists():
-            self.downloader.download_pf()
-        msg = (
-            f"{pf_path} does not exist. "
-            + "Make sure data files are available for specified fields."
+        self.prior = SDSSPrior(
+            sdss_dir, self.image_ids(), self, self.bands, prior_flux_ref_band, prior_config
         )
-        assert Path(pf_path).exists(), msg
-        self.pf_fits = fits.getdata(pf_path)
+        self.background = ImageBackground(items=self, bands=self.bands)
+        self.psf = SDSSPointSpreadFunction(sdss_dir, self.image_ids(), self.bands, psf_config)
 
-        fieldnums = self.pf_fits["FIELD"]
-        fieldgains = self.pf_fits["GAIN"]
-
-        # get desired field
-        for i, field in enumerate(fieldnums):
-            gain = fieldgains[i]
-            if (not self.fields) or field in self.fields:
-                self.rcfgcs.append((self.run, self.camcol, field, gain))
-        self.items = [None for _ in range(len(self.rcfgcs))]
+    def prepare_data(self):
+        for rcf_conf in self.sdss_fields:
+            run, camcol, fields = rcf_conf["run"], rcf_conf["camcol"], rcf_conf["fields"]
+            downloader = SDSSDownloader(run, camcol, fields[0], download_dir=str(self.sdss_path))
+
+            pf_file = f"photoField-{run:06d}-{camcol:d}.fits"
+            camcol_path = self.sdss_path.joinpath(str(run), str(camcol))
+            pf_path = camcol_path.joinpath(pf_file)
+            if not Path(pf_path).exists():
+                downloader.download_pf()
+            msg = (
+                f"{pf_path} does not exist. "
+                + "Make sure data files are available for specified fields."
+            )
+            assert Path(pf_path).exists(), msg
+            pf_fits = fits.getdata(pf_path)
+
+            assert pf_fits is not None, f"Could not load fits file {pf_path}."
+            fieldnums = pf_fits["FIELD"]  # type: ignore
+            fieldgains = pf_fits["GAIN"]  # type: ignore
```

Comment:
Are these `type: ignore` statements necessary? (We're not using mypy any more.)

Reply:
Done.